### PR TITLE
mcs: Remove domain time check from preemptionPoint

### PR DESCRIPTION
**Pull request by xurtis**
This removes the operations that trigger a reschedule or reprogram the timer from `preemptionPoint` to ensure they relevant state updates in the proof occur where it they are easier to verify.

### CODE REVIEW

**Review by michaelmcinerney — APPROVED** [review]
Looks great. Thanks Curtis!

File: include/kernel/thread.h [hunk 1]
```diff
@@ -229,18 +229,23 @@ static inline void updateTimestamp(void)
...
-        }
     }
+}
 
+/*
+ * Check if domain time has expired
+ */
+static inline void checkDomainTime(void)
+{
+    if (unlikely(CONFIG_NUM_DOMAINS > 1 && isCurDomainExpired())) {
```
Comment (michaelmcinerney) [inline include/kernel/thread.h:245]: `CONFIG_NUM_DOMAINS > 1` is already included in `isCurDomainExpired`, so this part can be removed

**Comment by michaelmcinerney** [comment]
Would we also like to change the guard in `scheduleChooseNewThread` to `isCurDomainExpired`?

**Comment by michaelmcinerney** [comment]
I've shown with this spec that we do indeed have nonzero domain time on kernel exit. It was very straightfoward with these changes. There are still most of the proofs for the scheduler invariants to go, but I might come back to those later

**Comment by lsf37** [comment]
Closing in favour of #505

**Pull request closed without merging**